saagieapi/projects/saagie_api.py

```diff
@@ -278,6 +326,56 @@ def delete_project_env_var(self, project_id, name):
         query = gql(gql_delete_env_var.format(project_env_id))
         return self.client.execute(query)
 
+
+    def update_project_env_var(self, project_id, name, new_name=None, value=None, description=None, is_password=None):
+        """
+        Update environment variable with provided function vairables if it exists
+        Parameters
+        ----------
+        project_id : str
+            ID of the project
+        name : str
+            Name of the environment to upgrade
+        new_name : str, optional
+            New name of the environment variable. If none provided, keep the actual one
+        value: str, optional
+            New value of the environment variable. If none provided, keep the actual one
+        description; str, optional
+            New description of the environment variable. If none provided, keep the actual one           
+        is_password: boolean, optional
+            New password boolean status. If none provided, keep the actual one
+        Returns
+        -------
+        None
+        """   
+
+        existing_env_var = self.get_project_env_vars(project_id)['projectEnvironmentVariables']
+     
+        if name not in [env_var['name'] for env_var in existing_env_var]:
+            raise ValueError("Environment variable does not exists")
+        
+        current_env_var = [d for d in existing_env_var if d['name'] == name][0]
+
+        if new_name:
+            current_env_var['name']=new_name
+        if value:
+            current_env_var['value']=value
+        if description:
+            current_env_var['description']=description
+        if is_password:
+            current_env_var['isPassword']=is_password
+
+
+        query = gql(gql_update_project_env_var.format(project_id,
+                                              current_env_var['id'],
+                                              current_env_var['name'], 
+                                              current_env_var['value'],
+                                              current_env_var['description'],
+                                              str(current_env_var['isPassword']).lower()))
+
+        return self.client.execute(query)
```

Comment:
is_password: when it is set to False: it will use the previous value

When the previous value of env var is a password, you cannot change only others parameters (you have to change the value)

To use variables in gql, you can do something like :


```
params = [d for d in existing_env_var if d['name'] == name][0]
        params["entityId"] = project_id

        if new_name:
            params['name']=new_name
        if value:
            params['value']=value
        if description:
            params['description']=description
        if is_password == True:
            params['isPassword']=is_password
        elif is_password == False:
            params['isPassword']=is_password


        query = gql(gql_update_env_var)

        return self.client.execute(query, variable_values=params)
```